backend/migrations/dataall_migrations/migrationmanager.py

```diff
@@ -68,6 +68,7 @@ def upgrade(self, target_revision_id=None):
         logger.info(f"Upgrade from {self.current_migration.revision_id()} to {target_revision_id or 'latest'}")
         executed_upgrades: Deque[Union[Type[MigrationBase], None]] = deque()
         saved_start_migration = self.current_migration
+        self.current_migration = self.current_migration.next_migration()
```

Comment:
I think this will make the following fail although it's valid
```
mm = Manager('0', Initial)
mm.upgrade()
mm.downgrade()
```
That's because `previous_migrations` won't contain the InitialMigration

Also I think for the same reason...
```
    manager = MigrationManager('1', m0)
    new_version = manager.upgrade()
    m0.up.assert_not_called()
    m1.up.assert_not_called()
    m2.up.assert_called_once()
    m3.up.assert_called_once()
    manager.downgrade()
    m1.down.assert_not_called()
    m2.down.assert_called_once() <--called twice
    m3.down.assert_called_once() <--called twice
```